internal/controller/authproxyworkload_controller_test.go

```diff
@@ -232,6 +233,92 @@ func TestReconcileState33(t *testing.T) {
 
 }
 
+func TestReconcileDeleteUpdatesWorkload(t *testing.T) {
+	resource := testhelpers.BuildAuthProxyWorkload(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "project:region:db")
+	resource.Generation = 1
+	resource.Finalizers = []string{finalizerName}
+	resource.Spec.Workload = v1alpha1.WorkloadSelectorSpec{
+		Kind: "Deployment",
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"app": "things"},
```

Comment:
Should we pull these labels out into a variable to make it clear that they're used below in the Deployment?